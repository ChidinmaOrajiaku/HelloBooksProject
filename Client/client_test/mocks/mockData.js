import { updatePassword } from "../../actions/updatePassword";

const mockData = {
  userDetails: {
    message: 'Welcome admin',
    username: 'admin',
    token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6ImRhZGR5Iiwicm9sZSI6InVzZXIiLCJpZCI6MiwiaWF0IjoxNTEzODc1ODE3LCJleHAiOjE1MTM5NjIyMTd9.oexUFhi08vhMU88LmZJvRDy9JFdThMpfb3Qz9DWWzgk'
  },
  userDetailsResponse: {
    username: 'daddy',
    role: 'user',
    id: 2,
    iat: 1513875817,
    exp: 1513962217
  },
  userCountResponse: {
    count: 7
  },
  borrowBookData: {
    bookId: 1,
    userId: 1,
  },
  borrowBookDataResponse: {
    returned: false,
    id: 24,
    userId: 1,
    bookId: 7,
    toReturnDate: '2018-01-14',
    updatedAt: '2017-12-21T21:16:08.442Z',
    createdAt: '2017-12-21T21:16:08.442Z',
    returnDate: null,
    categoryId: null
  },
  createBookData: {
    title: 'a',
    author: 'a',
    category: 'a',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1511133007/g2exfx6sgbymszspybts.jpg',
    review: 'Nice'
  },
  createBookErrorData: {
    author: 'a',
    category: 'a',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1511133007/g2exfx6sgbymszspybts.jpg',
    review: 'Nice'
  },
  createBookErrorResponse: 'An error occured',
  createBookResponse: {
    message: 'Succesfully added'
  },
  getABookData: {
    id: 7,
    title: 'Rage of Angels',
    author: 'Sidney Sheldon',
    category: 'Fiction',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1513096198/prbdr6eghnnrcewwpodf.jpg',
    review: 'Nice Book',
    createdAt: '2017-12-12T11:11:10.823Z',
    updatedAt: '2017-12-12T16:30:00.352Z',
    userId: null,
    categoryId: null
  },
  getAllBooksData: {
    id: 7,
    title: 'Rage of Angels',
    author: 'Sidney Sheldon',
    category: 'Fiction',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1513096198/prbdr6eghnnrcewwpodf.jpg',
    review: 'Nice Book',
    createdAt: '2017-12-12T11:11:10.823Z',
    updatedAt: '2017-12-12T16:30:00.352Z',
    userId: null,
    categoryId: null
  },
  getAllBorrowedBooksData: {
    id: 7,
    title: 'Rage of Angels',
    author: 'Sidney Sheldon',
    category: 'Fiction',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1513096198/prbdr6eghnnrcewwpodf.jpg',
    review: 'Nice Book',
    createdAt: '2017-12-12T11:11:10.823Z',
    updatedAt: '2017-12-12T16:30:00.352Z',
    userId: null,
    categoryId: null
  },
  getCategoryData: {
    category: 'Fiction'
  },
  getUserData: {
    firstname: 'daddy',
    lastname: 'daddy',
    username: 'daddy',
    password: 'daddy',
    email: 'daddy@gmail.com'
  },
  bookData: {
    title: 'a',
    author: 'b',
    category: 'a',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1511133007/g2exfx6sgbymszspybts.jpg',
    review: 'Nice'
  },
  returnBookData: {
    message: 'Successfully Returned'
  },
  userData: {
    verifyPassword: '1234',
    password: '4321'
  },
  errorMessage: 'An error occurred',
  image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1515403427/ena7dhy6xriztom6vuvo.jpg'
};

export default mockData;
