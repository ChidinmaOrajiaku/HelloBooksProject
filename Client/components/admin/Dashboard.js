import React from 'react';
import ReactDOM from 'react-dom';
import Footer from '../Footer';
import NavigationBar from '../NavigationBar';
import { connect } from 'react-redux';
import { adminCountBooksRequest, adminCountRentedBooksRequest } from '../../actions/booksAction';
import { adminCountUserRequest } from '../../actions/profileAction';

/**
 * @class Dashboard
 */
class Dashboard extends React.Component {
  /**
     * @constructor
     * @param {object} props 
     */
  constructor(props) {
    super(props);
    this.state = {
      loading: true,
      booksData: 0,
      rentedBooksData: 0,
      usersCountData: 0,
    };
  }

  /**
   * 
   * @constructor
   * @memberof Dashboard
   */
  componentDidMount() {
    this.props.adminCountBooksRequest().then(
      this.setState({
        booksData:localStorage.getItem('countBooks')
      })
    );
    this.props.adminCountRentedBooksRequest().then(
      this.setState({
        rentedBooksData:localStorage.getItem('countRentedBooks')
      })
    );
    this.props.adminCountUserRequest().then(
      this.setState({
        usersCountData:localStorage.getItem('countUsers')
      })
    );
  }

  /**
   * 
   * 
   * @param {any} nextProps 
   * @memberof Dashboard
   */

  /**
   * @constructor
   * @param {object} render
   */
  render() {
    return (
      <div className="adminDashboard container">
        <div className=""> <NavigationBar /> </div>
        <div className="row initial">
          <div className="col s6 m6">
            <div className="card">
              <div className="card-content text-center teal-text">
                <span className="card-title text-center">Total Books</span>
                <p>{this.state.booksData}</p>
              </div>
              <div className="card-action">
                <a href="/library" className="cardAction">View</a>
              </div>
            </div>
          </div>
          <div className="col s6 m6 black-text">
            <div className="card">
              <div className="card-content text-center teal-text">
                <span className="card-title text-center">Total Borrowed Books</span>
                <p>{this.state.rentedBooksData}</p>
              </div>
              <div className="card-action">
                <a className="cardAction" href="/library">View</a>
              </div>
            </div>
          </div>
          <div className="col s6 m6">
            <div className="card">
              <div className="card-content teal-text text-center">
                <span className="card-title text-center">Total Users</span>
                <p>{ this.state.usersCountData }</p>
              </div>
              <div className="card-action">
                <a className="cardAction" href="/history">View</a>
              </div>
            </div>
          </div>
        </div>
        <div className="row second">

        </div>
        <div> <Footer/></div>
      </div>
    );
  }
}

const mapStateToProps = state => (
  {
    booksData: state.books.books,
    rentedBooksData: state.books.rentedBooks,
    usersCountData: state.userState.adminCountUsers
  }
);

export default connect(mapStateToProps, { adminCountBooksRequest, adminCountRentedBooksRequest, adminCountUserRequest })(Dashboard);
