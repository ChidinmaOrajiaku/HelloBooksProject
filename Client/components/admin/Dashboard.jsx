import React from 'react';
import ReactDOM from 'react-dom';
import { connect } from 'react-redux';
import Footer from '../Footer';
import NavigationBar from '../NavigationBar';
import {
  adminCountBooksRequest,
  adminCountRentedBooksRequest,
  adminCountNotReturnedBooksRequest,
  adminCountCategoryRequest,
  adminCreateCategoryRequest
} from '../../actions/booksAction';
import { adminCountUserRequest } from '../../actions/adminCountUsers';

/**
 * @class Dashboard
 */
export class Dashboard extends React.Component {
  /**
     * @constructor
     * @param {object} props
     */
  constructor(props) {
    super(props);
    this.state = {
      loading: true,
      booksData: '',
      rentedBooksData: '',
      usersCountData: '',
      categoryCountData: '',
      category: ''
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleCategory = this.handleCategory.bind(this);
  }

  /**
   *
   * @returns {object} mounted components
   * @memberof Dashboard
   */
  componentDidMount() {
    this.props.adminCountBooksRequest();
    this.props.adminCountRentedBooksRequest();
    this.props.adminCountUserRequest();
    this.props.adminCountNotReturnedBooksRequest();
    this.props.adminCountCategoryRequest();
    $('.modal').modal({
      dismissible: true,
      opacity: 0.5,
      inDuration: 300,
      outDuration: 200,
      startingTop: '2%',
      endingTop: '20%',
    });
  }

  /**
   *
   * @returns {nextProps} next props
   * @param {any} nextProps
   * @memberof Dashboard
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.booksData[0].bookCount) {
      this.setState({
        booksData: nextProps.booksData[0].bookCount,
      });
    }
    if (nextProps.booksData[0].rentedBookCount) {
      this.setState({
        rentedBooksData: nextProps.booksData[0].rentedBookCount,
      });
    }
    if (nextProps.booksData[0].categoryCount) {
      this.setState({
        categoryCountData: nextProps.booksData[0].categoryCount,
      });
    }
    if (nextProps.usersCountData[0]) {
      this.setState({
        usersCountData: nextProps.usersCountData[0].adminCountUsers,
      });
    }
  }

  /**
 *
 * @param {any} event
 * @memberof AddBooks
 * @returns {object} SyntheticEvent
 */
  handleChange(event) {
    this.setState({ [event.target.id]: event.target.value });
  }

  /**
 *
 * @returns {event} handles category change
 * @param {any} event
 * @memberof Dashboard
 */
  handleCategory(event) {
    event.preventDefault();
    this.props.adminCreateCategoryRequest(this.state);
    setTimeout(() => {
      if (this.props.booksData[0].isCreated === true) {
        Materialize.toast('Successfully Added', 2000, 'teal rounded');
      } else {
        Materialize.toast('Not Created', 2000, 'red rounded');
      }
    }, 2000);
  }

  /**
   *
   *
   * @memberof Dashboard
   * @returns {object} ReactMarkupElement
   */
  render() {
    return (
      <div className="adminDashboard ">
        <div className="container row">
          <div className=""> <NavigationBar /> </div>
          <h4 className="col s12 m9 offset-m6 offset-s2 white-text"> DASHBOARD </h4>
          <div className="row initial">
            <div className="col s12 m6 pull-s3">
              <div className="card">
                <div className="card-content text-center teal-text">
                  <span className="card-title text-center">Total Books</span>
                  <p>{this.state.booksData}</p>
                </div>
                <div className="card-action">
                </div>
              </div>
            </div>
            <div className="col s12 m6 pull-s3 black-text">
              <div className="card">
                <div className="card-content text-center teal-text">
                  <span className="card-title text-center">Total Borrowed Books
                  </span>
                  <p>{this.state.rentedBooksData}</p>
                </div>
                <div className="card-action">
                </div>
              </div>
            </div>
            <div className="col s12 m6 pull-s3">
              <div className="card">
                <div className="card-content teal-text text-center">
                  <span className="card-title text-center">Total Users</span>
                  <p>{ this.state.usersCountData }</p>
                </div>
                <div className="card-action">

                </div>
              </div>
            </div>
            <div className="col s12 m6 pull-s3">
              <div className="card">
                <div className="card-content teal-text text-center">
                  <span className="card-title text-center">Total Category</span>
                  <p>{ this.state.categoryCountData }</p>
                </div>
                <div className="card-action">
                  <a href="#modal1" className="modal-trigger white-text">Create</a>
                </div>
              </div>
            </div>
            <div id="modal1" className="modal">
              <div className="modal-content">
                <div className="row">
                  <div className="input-field col s12">
                    <input
                      value={this.state.category}
                      onChange={this.handleChange}
                      id="category" type="text"
                      className="validate"
                      required="required"
                    />
                    <label htmlFor="category">Category</label>
                  </div>
                </div>
                <div className="modal-footer">
                  <button
                    onClick={this.handleCategory}
                    type="submit"
                    className="createButton">Create</button>
                  <a className="modal-action modal-close">
                    <button
                    className="cancelButton"
                    id="cancelButton">Cancel
                    </button>
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div> <Footer /></div>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => (
  {
    booksData: state.books,
    usersCountData: state.userState,
    createCategory: state.books
  }
);

export default connect(
  mapStateToProps,
  {
    adminCountBooksRequest,
    adminCountRentedBooksRequest,
    adminCountUserRequest,
    adminCountNotReturnedBooksRequest,
    adminCreateCategoryRequest,
    adminCountCategoryRequest
  }
)(Dashboard);
