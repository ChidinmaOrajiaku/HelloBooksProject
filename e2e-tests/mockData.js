module.exports = {
  file: {
    lastModified: 1508779018000,
    lastModifiedDate: 'Mon Oct 23 2017 18:16:58 GMT+0100 (WAT)',
    name: '3 card game.jpg',
    size: 642971,
    type: 'image/jpeg',
    webkitRelativePath: ''
  },
};

