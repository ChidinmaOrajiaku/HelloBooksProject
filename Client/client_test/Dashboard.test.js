import React from 'react';
import $ from 'jquery';
import toJson from 'enzyme-to-json';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import mockData from './mocks/mockData';
import { Dashboard } from '../components/admin/Dashboard.jsx';

configure({ adapter: new Adapter() });

describe('<Dashboard />', () => {
  const props = {
    adminCreateCategoryRequest: jest.fn(),
    adminCountBooksRequest: jest.fn(),
    adminCountRentedBooksRequest: jest.fn(),
    adminCountNotReturnedBooksRequest: jest.fn(),
    adminCountCategoryRequest: jest.fn(),
    adminCountUserRequest: jest.fn()
  };
  let mountedComponent;
  const DashboardItem = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<Dashboard {...props}/>);
    }
    return mountedComponent;
  };

  it('should call handleCategory method', () => {
    const spy = jest.spyOn(DashboardItem().instance(), 'handleCategory');
    const event = {
      preventDefault: jest.fn(),
      target: {
        value: 'abd',
      }
    };
    DashboardItem().instance().handleCategory(event);
    expect(spy).toHaveBeenCalled();
  });

  it('should call handleChange method', () => {
    const spy = jest.spyOn(DashboardItem().instance(), 'handleChange');
    const event = {
      preventDefault: jest.fn(),
      target: {
        id: 'review',
        value: 'chideberecom'
      }
    };
    DashboardItem().instance().handleChange(event);
    expect(spy).toHaveBeenCalled();
  });

  it('should match snapshot test', () => {
    const component = DashboardItem();
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
