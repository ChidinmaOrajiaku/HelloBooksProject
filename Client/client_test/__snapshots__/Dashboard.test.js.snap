// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> should match snapshot test 1`] = `
<div
  className="adminDashboard"
>
  <div
    className="container"
  >
    <div
      className=""
    >
       
      <Connect(withRouter(NavigationBar)) />
       
    </div>
    <div
      className="row initial"
    >
      <div
        className="col s6 m6"
      >
        <div
          className="card"
        >
          <div
            className="card-content text-center teal-text"
          >
            <span
              className="card-title text-center"
            >
              Total Books
            </span>
            <p />
          </div>
          <div
            className="card-action"
          />
        </div>
      </div>
      <div
        className="col s6 m6 black-text"
      >
        <div
          className="card"
        >
          <div
            className="card-content text-center teal-text"
          >
            <span
              className="card-title text-center"
            >
              Total Borrowed Books
            </span>
            <p />
          </div>
          <div
            className="card-action"
          />
        </div>
      </div>
      <div
        className="col s6 m6"
      >
        <div
          className="card"
        >
          <div
            className="card-content teal-text text-center"
          >
            <span
              className="card-title text-center"
            >
              Total Users
            </span>
            <p />
          </div>
          <div
            className="card-action"
          />
        </div>
      </div>
      <div
        className="col s6 m6"
      >
        <div
          className="card"
        >
          <div
            className="card-content teal-text text-center"
          >
            <span
              className="card-title text-center"
            >
              Total Category
            </span>
            <p />
          </div>
          <div
            className="card-action"
          >
            <a
              className="modal-trigger white-text"
              href="#modal1"
            >
              Create
            </a>
          </div>
        </div>
      </div>
      <div
        className="modal"
        id="modal1"
      >
        <div
          className="modal-content"
        >
          <div
            className="row"
          >
            <div
              className="input-field col s12"
            >
              <input
                className="validate"
                id="category"
                onChange={[Function]}
                required="required"
                type="text"
                value=""
              />
              <label
                htmlFor="category"
              >
                Category
              </label>
            </div>
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="createButton"
              onClick={[Function]}
              type="submit"
            >
              Create
            </button>
            <a
              className="modal-action modal-close"
            >
              <button
                className="cancelButton"
                id="cancelButton"
              >
                Cancel
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div>
       
      <Component />
    </div>
  </div>
</div>
`;
